File: frontend/tic-tac-toe-ui/src/types/game.ts
export interface Move {
  moveNumber:number;
  player:string;
  row:number;
  column:number;
}

export interface Game {
  id:string;
  board:string[];
  currentPlayer:string;
  status:string;
  winner:string;
  winningCells:number[];
  moveHistory:Move[];
}

File: TicTacToe.Api/Models/Game.cs
using TicTacToe.Api.Enums;

namespace TicTacToe.Api.Models;

public class Game
{
    public Guid Id { get; set; }

    public string[] Board { get; set; }
        = new string[9];

    public string CurrentPlayer { get; set; }
        = "X";

    public GameMode Mode { get; set; }

    public GameStatus Status { get; set; }
        = GameStatus.InProgress;

    public string? Winner { get; set; }

    public List<int> WinningCells { get; set; }
        = new();

    public List<Move> MoveHistory { get; set; }
        = new();
}